src/gc.c

```diff
@@ -1856,14 +1856,37 @@ STATIC_INLINE uintptr_t gc_read_stack(void *_addr, uintptr_t offset,
     return *(uintptr_t*)real_addr;
 }
 
-JL_NORETURN NOINLINE void gc_assert_datatype_fail(jl_ptls_t ptls, jl_datatype_t *vt,
-                                                  jl_gc_markqueue_t *mq) JL_NOTSAFEPOINT
+#define GC_ASSERT_PARENT_VALIDITY
```

Comment:
@d-netto this is you setting this option to enable the flag, right?

Please separate that out, so that this commit is **only** the backport of the _option,_ and then we should have a separate commit that actually _enables_ the option. That way, we could drop this patch when upgrading to 1.10, but keep the patch that enables the feature.

Can you put this in a separate commit, and also in a separate file? I think this should be in our Make.user or something like that, right? Or in a nix file? Is there a place where we aggregate these build flags?

Reply:
Yes. This option was set by me: I just want to run some benchmarks to see what is the performance penalty of doing that, though I plan to disable it before merging. See [note](https://github.com/RelationalAI/julia/pull/53#issuecomment-1723755126).